core/src/main/java/org/elasticsearch/index/engine/SegmentsStats.java

```diff
@@ -266,17 +319,16 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         static final XContentBuilderString VERSION_MAP_MEMORY_IN_BYTES = new XContentBuilderString("version_map_memory_in_bytes");
         static final XContentBuilderString FIXED_BIT_SET = new XContentBuilderString("fixed_bit_set");
         static final XContentBuilderString FIXED_BIT_SET_MEMORY_IN_BYTES = new XContentBuilderString("fixed_bit_set_memory_in_bytes");
+        static final XContentBuilderString MEMORY_RESOURCES = new XContentBuilderString("memory_resources");
+        static final XContentBuilderString DISK_RESOURCES = new XContentBuilderString("disk_resources");
     }
 
     @Override
     public void readFrom(StreamInput in) throws IOException {
         count = in.readVLong();
```

Comment:
I think we need to check the incoming version and have back-compat here, i.e. if the version is < X (X = first version when we add this feature), we read the old (only memory) stats.